homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -566,39 +558,117 @@ def get_entity_state(config, entity):
             data[STATE_HUE] = 0
             data[STATE_SATURATION] = 0
 
+    # Clamp brightness, hue, saturation, and color temp to valid values
+    for (key, v_min, v_max) in (
+        (STATE_BRIGHTNESS, HUE_API_STATE_BRI_MIN, HUE_API_STATE_BRI_MAX),
+        (STATE_HUE, HUE_API_STATE_HUE_MIN, HUE_API_STATE_HUE_MAX),
+        (STATE_SATURATION, HUE_API_STATE_SAT_MIN, HUE_API_STATE_SAT_MAX),
+        (STATE_COLOR_TEMP, HUE_API_STATE_CT_MIN, HUE_API_STATE_CT_MAX),
+    ):
+        if data[key] is not None:
+            data[key] = max(v_min, min(data[key], v_max))
+
     return data
 
 
-def entity_to_json(config, entity, state):
+def entity_to_json(config, entity):
     """Convert an entity to its Hue bridge JSON representation."""
     entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
-    if (entity_features & SUPPORT_BRIGHTNESS) or entity.domain != light.DOMAIN:
-        return {
-            "state": {
-                HUE_API_STATE_ON: state[STATE_ON],
-                HUE_API_STATE_BRI: state[STATE_BRIGHTNESS],
-                HUE_API_STATE_HUE: state[STATE_HUE],
-                HUE_API_STATE_SAT: state[STATE_SATURATION],
-                "reachable": entity.state != STATE_UNAVAILABLE,
-            },
-            "type": "Dimmable light",
-            "name": config.get_entity_name(entity),
-            "modelid": "HASS123",
-            "uniqueid": entity.entity_id,
-            "swversion": "123",
-        }
-    return {
+    unique_id = hashlib.md5(entity.entity_id.encode()).hexdigest()
+    unique_id = "00:%s:%s:%s:%s:%s:%s:%s-%s" % (
```

Comment:
Please use f-strings instead of old style string formatting.

Reply:
Here I would actually just do a `":".join(("00", val1, val2, …))`

Reply:
> Here I would actually just do a `":".join(("00", val1, val2, …))`

While that would work, there's actually a hyphen before the last two hex characters so that would have to be broken out specifically. As such, there's not a "clean" way to represent it with `join()`.

With `join()`:
```
unique_id = ":".join((
    "00",
    unique_id[0:2],
    unique_id[2:4],
    unique_id[4:6],
    unique_id[6:8],
    unique_id[8:10],
    unique_id[10:12],
    unique_id[12:14],
)) + "-" +  unique_id[14:16]
```
**Pro:** Much easier to read and modify
**Con:** Concatenation on top of a join

With f-strings that "problem" goes away, but the string overall is harder to read and prone to causing confusion for editors in the future:
```
# "Plain" - don't care about string length
unique_id_fstring = f"00:{unique_id[0:2]}:{unique_id[2:4]}:{unique_id[4:6]}:{unique_id[6:8]}:{unique_id[8:10]}:{unique_id[10:12]}:{unique_id[12:14]}-{unique_id[14:16]}"

# "Tidy" - don't use lines longer than 80 characters results in:
unique_id_fstring2 = f"00:{unique_id[0:2]}:{unique_id[2:4]}:{unique_id[4:6]}:" \
                     f"{unique_id[6:8]}:{unique_id[8:10]}:{unique_id[10:12]}:" \
                     f"{unique_id[12:14]}-{unique_id[14:16]}"
```
**Pro:** Uses f-strings (ooohhh, shiny)
**Con:** A bit harder to read

Either of you guys care to chime in? I didn't make any change yet and to be honest, I'm not entirely impressed with either solution.

Reply:
okay, in that case let's use 

```suggestion
    unique_id = "00:%s:%s:%s:%s:%s:%s:%s-%s".format(
```

Anything but `%` :)

Reply:
@balloob Done, committed. Thanks!